tests/ast/test_pre_parser.py

```diff
@@ -1,25 +1,53 @@
 import pytest
-from pkg_resources.extern import packaging
 
-from vyper import __version__
 from vyper.ast.pre_parser import validate_version_pragma
 from vyper.exceptions import VersionException
 
 SRC_LINE = (1, 0)  # Dummy source line
-version = packaging.version.Version(__version__)._version
-bumped_major_version = f"{version.release[0]+1}.0.0"
-bumped_minor_version = f"{version.release[0]}.{version.release[1]+1}.0"
-bumped_patch_version = f"{version.release[0]}.{version.release[1]}.{version.release[2]+1}"
-invalid_version = "0.1"
+COMPILER_VERSION = "0.1.1"
+PRERELEASE_COMPILER_VERSION = "0.1.1b7"
 
 
-@pytest.mark.parametrize("file_version", [bumped_patch_version, __version__])
-def test_valid_version_pragma(file_version):
+@pytest.fixture
+def mock_version(monkeypatch):
+    def set_version(version):
+        monkeypatch.setattr("vyper.__version__", version)
+    return set_version
+
+
+valid_versions = ["0.1.1", ">0.0.1", "^0.1.0", "<=1.0.0 >=0.1.0", "0.1.0 - 1.0.0", "~0.1.0", "0.1",
```

Comment:
Apparently, `0.1.x`, `0.x`, and `x` should also work?

https://docs.npmjs.com/about-semantic-versioning#using-semantic-versioning-to-specify-update-types-your-package-can-accept

Reply:
Okay, I'll add those test cases.